spot_driver/launch/spot_image_publishers.launch.py

```diff
@@ -0,0 +1,260 @@
+        )
+    return composable_node_descriptions
+
+
+def create_point_cloud_nodelets(
+    context: launch.LaunchContext,
+    spot_name: LaunchConfiguration,
+    has_arm: bool,
+) -> List[launch_ros.descriptions.ComposableNode]:
+    """Create the list of depth_image_proc::PointCloudXyzrgbNode composable nodes required to generate point clouds for
+    each pair of RGB and registered depth cameras."""
+
+    composable_node_descriptions = []
+
+    for camera in get_camera_sources(has_arm):
+        composable_node_descriptions.append(
+            launch_ros.descriptions.ComposableNode(
+                package="depth_image_proc",
+                plugin="depth_image_proc::PointCloudXyzrgbNode",
+                name="point_cloud_xyzrgb_node_" + camera,
+                namespace=spot_name,
+                # Each entry in the remappings list is a tuple.
+                # The first element in the tuple is the internal name of the topic used within the nodelet.
+                # The second element is the external name of the topic used by other nodes in the system.
+                remappings=[
+                    ("rgb/camera_info", PathJoinSubstitution(["camera", camera, "camera_info"]).perform(context)),
+                    ("rgb/image_rect_color", PathJoinSubstitution(["camera", camera, "image"]).perform(context)),
+                    (
+                        "depth_registered/image_rect",
+                        PathJoinSubstitution(["depth_registered", camera, "image"]).perform(context),
+                    ),
+                    ("points", PathJoinSubstitution(["depth_registered", camera, "points"]).perform(context)),
+                ],
+            ),
+        )
+    return composable_node_descriptions
+
+
+def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
+    config_file = LaunchConfiguration("config_file")
+    spot_name = LaunchConfiguration("spot_name").perform(context)
+    depth_registered_mode_config = LaunchConfiguration("depth_registered_mode")
+    publish_point_clouds_config = LaunchConfiguration("publish_point_clouds")
+    mock_enable = IfCondition(LaunchConfiguration("mock_enable", default="False")).evaluate(context)
+
+    # if config_file has been set (and is not the default empty string) and is also not a file, do not launch anything.
+    config_file_path = config_file.perform(context)
+    if (config_file_path != "") and (not os.path.isfile(config_file_path)):
+        raise FileNotFoundError("Configuration file '{}' does not exist!".format(config_file_path))
+
+    if mock_enable:
+        mock_has_arm = IfCondition(LaunchConfiguration("mock_has_arm")).evaluate(context)
+        has_arm = mock_has_arm
+    else:
+        has_arm = spot_has_arm(config_file_path=config_file.perform(context), spot_name=spot_name)
+
+    depth_registered_mode_string = depth_registered_mode_config.perform(context).lower()
+    depth_registered_mode = DepthRegisteredMode(depth_registered_mode_string)
```

Comment:
What happens if the string is not one of the possible values? Does it throw an exception?

Reply:
It would throw an exception, but right now that's not possible because the `depth_registered_mode` launch argument is restricted to the same values of the `DepthRegisteredMode` enum. Not sure if there's a cleaner way to handle this